### PR TITLE
Not logging command arguments and environment when logging config [full ci]

### LINKED ISSUE #1659: Not enough information in build logs to debug container issues
Currently only the log from the container and vmware.log are present in the build log for a failed build. This currently does not give enough information, for instance, to debug networking issues. We need a way get more information about the container in the build logs. Maybe a dump of guestinfo, and other commands like we do for the VCH.

### PR DESCRIPTION
**Pull request by hmahmood**
Fixes #1659

### CODE REVIEW

File: lib/tether/tether.go [hunk 1]
```diff
@@ -584,14 +582,37 @@ func logConfig(config *ExecutorConfig) {
...
+			if _, ok := keys[k]; !ok {
+				sink[k] = v
+			}
+
+			return nil
+		},
+		config,
+	)
+
+	for k, v := range sink {
```
Comment (hickeng) [inline lib/tether/tether.go]: Why not log directly in the func passed to Encode?

Comment (hmahmood) [inline lib/tether/tether.go]: The logging is in there is affected by the log level in the `extraconfig` package, which could be different from what the tether is using.

File: pkg/vsphere/extraconfig/keys.go [hunk 2]
```diff
@@ -262,3 +262,114 @@ func addPrefixToKey(header, prefix, key string) string {
...
+	log.Debugf("v=%#v, field=%#v, prefix=%#v", v, field, prefix)
+	if v.Kind() == reflect.Ptr {
+		return calculateKeys(v.Elem(), field, prefix)
+	}
+
+	if field == "" {
+		return []string{prefix}
+	}
+
+	s := strings.SplitN(field, ".", 2)
```
Comment (fdawg4l) [inline pkg/vsphere/extraconfig/keys.go]: I'm new to this guest info stuff.  Is this a field separator inherent to extraconfig?  If so does this conflict or break with this (https://github.com/vmware/vic/pull/3062/files)?

**Comment by fdawg4l** [comment]
LGTM

**Comment by caglar10ur** [comment]
LGTM (would love to convert | to a const so that we can change it easily but that's for another PR)

**Comment by hmahmood** [comment]
.

**Comment by hmahmood** [comment]
♻️

**Pull request merged**